.travis.yml

```diff
@@ -25,41 +43,74 @@ install:
   - conda config --append channels conda-forge
   # Useful for debugging any issues with conda
   - conda info -a
-  # Test conda build
-  - conda build -q conda-recipe --python $TRAVIS_PYTHON_VERSION --output-folder bld-dir
-  - conda config --add channels "file://`pwd`/bld-dir"
-  #Grab all dependencies
-  - conda create -q -n test-environment python=$TRAVIS_PYTHON_VERSION pcdsdevices --file dev-requirements.txt
-  #Launch Conda environment
-  - source activate test-environment
+  # Test conda build and create test environment
+  - |
+    if [[ $UNIT_TEST || $BUILD_DOCS ]]; then
+      echo "Building full environment"
+      conda build -q conda-recipe --python $TRAVIS_PYTHON_VERSION --output-folder bld-dir
+      conda config --add channels "file://`pwd`/bld-dir"
+      conda create -q -n test-environment python=$TRAVIS_PYTHON_VERSION pcdsdevices --file dev-requirements.txt
+    elif [[ $STYLE ]]; then
+      echo "Building minimal flake8 environment"
+      conda create -q -n test-environment python=$TRAVIS_PYTHON_VERSION flake8
+    fi
+  # Launch Conda environment
+  - conda activate test-environment
+  # Add the docs packages
+  - |
+    if [[ $BUILD_DOCS ]]; then
+      echo "Adding docs requirements"
+      conda install --file docs-requirements.txt
+    fi
 
 script:
-  - coverage run run_tests.py
-  - coverage report -m
-  - flake8 pcdsdevices/*.py
-  - flake8 tests/*.py
-  - set -e
+  # Run tests
+  - |
+    if [[ $UNIT_TEST ]]; then
+      echo "Running tests"
+      coverage run run_tests.py
+      coverage report -m
+    fi
+  # Check style
+  - |
+    if [[ $STYLE ]]; then
+      echo "Checking style"
+      flake8 pcdsdevices/*.py
+      flake8 tests/*.py
+    fi
   # Build docs
-  - conda install --file docs-requirements.txt
-  - pushd docs
-  - make html
-  - popd
+  - |
+    if [[ $BUILD_DOCS ]]; then
+      echo "Building docs"
+      set -e
+      conda install --file docs-requirements.txt
+      pushd docs
+      make html
+      popd
+    fi
   # Upload docs
   - |
     if [[ -n "$DOCTR_DEPLOY_ENCRYPTION_KEY_PCDSHUB_PCDSDEVICES" && $BUILD_DOCS ]]; then
+      echo "Deploying docs"
       doctr deploy . --built-docs docs/build/html --deploy-branch-name gh-pages
     fi
 
 after_success:
-  - codecov
   - |
-    if [[ $TRAVIS_PULL_REQUEST == false && $TRAVIS_REPO_SLUG == $OFFICIAL_REPO && $TRAVIS_BRANCH == $TRAVIS_TAG  && $TRAVIS_TAG != '' && $CONDA_UPLOAD_TOKEN_TAG != '' ]]; then
+    if [[ $UPLOAD ]]; then
+      echo "Uploading to codecov"
+      codecov
+    fi
+  - |
+    if [[ $UPLOAD && $TRAVIS_PULL_REQUEST == false && $TRAVIS_REPO_SLUG == $OFFICIAL_REPO && $TRAVIS_BRANCH == $TRAVIS_TAG  && $TRAVIS_TAG != '' && $CONDA_UPLOAD_TOKEN_TAG != '' ]]; then
+      echo "Uploading to pcds_tag channel"
       export ANACONDA_API_TOKEN=$CONDA_UPLOAD_TOKEN_TAG
-      anaconda upload bld-dir/linux-64/*.tar.bz2
+      anaconda upload bld-dir/noarch/*.tar.bz2
```

Comment:
😄 :+1:

Reply:
If we're going to do this in a uniform way in the ci-helpers lib, should this line look like:
`anaconda upload bld-dir/*/*.tar.bz2`?